ProjectLighthouse/Helpers/CensorHelper.cs

```diff
@@ -19,80 +21,86 @@ public static class CensorHelper
+            {
+                Censor(censorIndices[i], profanity.Length, stringBuilder);
+                profaneCount += 1;
+            }
+        }
 
-    private static string Censor(int profanityIndex, int profanityLength, string message)
-    {
-        StringBuilder sb = new();
+        if (profaneCount > 0 && message.Length <= 94 && ServerConfiguration.Instance.LogChatFiltering) // 94 = lbp char limit
+            Logger.Info($"Censored {profaneCount} profane words from message \"{message}\"", LogArea.Filter);
 
-        sb.Append(message.AsSpan(0, profanityIndex));
+        return stringBuilder.ToString();
+    }
 
+    private static void Censor(int profanityIndex, int profanityLength, StringBuilder message)
+    {
         switch (CensorConfiguration.Instance.UserInputFilterMode)
         {
             case FilterMode.Random:
                 char prevRandomChar = '\0';
-                for (int i = 0; i < profanityLength; i++)
+                for (int i = profanityIndex; i < profanityIndex + profanityLength; i++)
                 {
-                    if (message[i] == ' ')
-                    {
-                        sb.Append(' ');
-                    }
-                    else
-                    {
-                        char randomChar = randomCharacters[CryptoHelper.GenerateRandomInt32(0, randomCharacters.Length)];
-                        if (randomChar == prevRandomChar) randomChar = randomCharacters[CryptoHelper.GenerateRandomInt32(0, randomCharacters.Length)];
-
-                        prevRandomChar = randomChar;
-                        sb.Append(randomChar);
-                    }
+                    if (char.IsWhiteSpace(message[i])) continue;
+                    
+                    char randomChar = randomCharacters[CryptoHelper.GenerateRandomInt32(0, randomCharacters.Length)];
+                    if (randomChar == prevRandomChar)
+                        randomChar = randomCharacters[CryptoHelper.GenerateRandomInt32(0, randomCharacters.Length)];
+
+                    prevRandomChar = randomChar;
+                    message[i] = randomChar;
                 }
 
                 break;
             case FilterMode.Asterisks:
-                for(int i = 0; i < profanityLength; i++)
+                for(int i = profanityIndex; i < profanityIndex + profanityLength; i++)
                 {
-                    sb.Append(message[i] == ' ' ? ' ' : '*');
+                    if (char.IsWhiteSpace(message[i])) continue;
+
+                    message[i] = '*';
                 }
 
                 break;
             case FilterMode.Furry:
+                // Might wanna remove this.
+                // With the optimizations (using StringBuilders), this can cause corruption if the furry word isn't the same length as the profane word.
+                // I'm too fucking lazy and have too much of a migraine to fix this.
+                // - Rosie
```

Comment:
This should be addressed. Either rewrite the function and make some unit tests or delete the furry mode.

Reply:
This was addressed in https://github.com/LBPUnion/ProjectLighthouse/pull/795/commits/5f1ab750709ee3cf5e10160b2806dad23f4d6d69